examples/example_spectral_filter.py

```diff
@@ -0,0 +1,215 @@
+            center: 1D control value between [0.0, 1.0] for the center frequency of the filter.
+            width: 1D control value between [0.0, 1.0] for the bandwidth of the filter.
+            amount: 1D control value between [0.0, 1.0] for the amount of attenuation.
+        """
+        if amount == 0.0:
+            return x
+        n_bins = x.size(1)  # Figure out how many bins we have to work with
+        # Find the center freq bin
+        center_bin_idx = self._map_0to1_val_to_log_bin_idx(center, n_bins - 1)
+        # Find the lowest freq bin
+        lo_bin_idx = self._map_0to1_val_to_log_bin_idx(center * (1.0 - width), n_bins - 1)
+        lo_bin_idx = max(0, lo_bin_idx)
+        # Find the highest freq bin
+        hi_bin_idx = self._map_0to1_val_to_log_bin_idx(center + ((1.0 - center) * width), n_bins - 1)
+        hi_bin_idx = min(n_bins - 1, hi_bin_idx)
+        # If the filter has 0 width, we don't need to do anything
+        if hi_bin_idx - lo_bin_idx == 0:
+            return x
+        # Filter the low bins of the notch
+        if center_bin_idx - lo_bin_idx > 0:
+            # Using a linear spacing here is not ideal since the frequency bins are not linearly spaced,
+            # but this is just an example
+            lo_filter = 1.0 - (tr.linspace(0.0, 1.0, center_bin_idx - lo_bin_idx + 2)[1:-1] * amount)
+            lo_filter = lo_filter.view(1, -1, 1)
+            x[:, lo_bin_idx:center_bin_idx, :] *= lo_filter
+        # Filter the high bins of the notch
+        if hi_bin_idx - center_bin_idx > 0:
+            # Using a linear spacing here is not ideal since the frequency bins are not linearly spaced,
+            # but this is just an example
+            hi_filter = 1.0 - (tr.linspace(1.0, 0.0, hi_bin_idx - center_bin_idx + 1)[:-1] * amount)
+            hi_filter = hi_filter.view(1, -1, 1)
+            x[:, center_bin_idx:hi_bin_idx, :] *= hi_filter
+        return x
+
+
+class SpectralFilterWrapper(WaveformToWaveformBase):
+    def __init__(self,
+                 spectral_filter_model: nn.Module,
+                 model_io_n_frames: int = 16,
+                 io_n_ch: int = 2,
+                 n_fft: int = 2048,
+                 hop_len: int = 512,
+                 fade_n_samples: int = 384,  # Cross-fade for 3/4 of the hop_len to ensure no buzzing in the wet audio
+                 use_debug_mode: bool = True) -> None:
+        """
+        Creates a modified WaveformToWaveformBase wrapper that can be used to create spectral neural audio effects.
+        Feel free to use this as a starting point to make your own spectral effects!
+
+        Args:
+            spectral_filter_model: a spectral model, in this example a filter (could be replaced with anything).
+            model_io_n_frames: the number of STFT frames the spectral model expects as input and output.
+            io_n_ch: the number of channels the spectral model expects as input and output (should be 1 or 2).
```

Comment:
Is there any reason you need to use this instead of the is input mono, is output mono methods? Looks like it's not enforced to be 1 or 2 outside debug mode.